File: algorithms/202.c
#include <stdio.h>
#include <stdbool.h>

bool isHappy(int n)
{
    while ( true )
    {
        if ( n == 1 )
            return true;
        if ( n == 4 )
            return false;
        int sum = 0;
        while ( n )
        {
            sum += (n%10)*(n%10);
            n /= 10;
        }
        n = sum;
    }
}

int main()
{
    isHappy(9);
    return 0;
}

File: algorithms/python/202.py
class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        m = set([n])
        while n != 1:
            n = sum(map(lambda x: int(x)**2, str(n)))
            if n in m:
                return False
            else:
                m.add(n)
        return True
